### ISSUE TITLE
ROS in Ubuntu to drive and record insta360 camera topics

### ISSUE THREAD

**Issue by gyt290811** [tu-darmstadt-ros-pkg/image_projection#9]
Hello, is there any way to use ROS in Ubuntu to drive and record insta360 camera topics? I saw that you have already recorded a bag file of the insta360 camera.

```
Getting started
Launching the demo
Make sure to open a new terminal after building the project for the first time.

Start by downloading the demo bag file with

roscd image_projection_demo/bags/
wget https://www.sim.informatik.tu-darmstadt.de/~oehler/insta360_air_robocup2018.bag
Launch the demo with

roslaunch image_projection_demo demo_insta360.launch
An [rqt](http://wiki.ros.org/rqt) window with the source video and projection examples will open. The [rqt_reconfigure](http://wiki.ros.org/rqt_reconfigure) GUI on the right side can be used to change projection parameters at runtime.
```

**Comment by Martin-Oehler** [comment]
Hello,
you can connect the Insta360 Air via USB to a ubuntu pc and it will be recognized as a UVC device. You can use any ros camera driver to stream the fisheye images. We use http://wiki.ros.org/libuvc_camera.

Best regards,
Martin

**Comment by Gatsby23** [comment]
> Hello, you can connect the Insta360 Air via USB to a ubuntu pc and it will be recognized as a UVC device. You can use any ros camera driver to stream the fisheye images. We use http://wiki.ros.org/libuvc_camera.
> 
> Best regards, Martin

What fps of this camera?

**Comment by Martin-Oehler** [comment]
The Insta360 Air supports the following formats:

```
    [0]: 'MJPG' (Motion-JPEG, compressed)
        Size: Discrete 1472x736
            Interval: Discrete 0.033s (30.000 fps)
        Size: Discrete 2176x1088
            Interval: Discrete 0.033s (30.000 fps)
        Size: Discrete 3008x1504
            Interval: Discrete 0.033s (30.000 fps)
    [1]: 'H264' (H.264, compressed)
        Size: Discrete 1920x960
            Interval: Discrete 0.033s (30.000 fps)
            Interval: Discrete 0.042s (24.000 fps)
            Interval: Discrete 0.050s (20.000 fps)
        Size: Discrete 2560x1280
            Interval: Discrete 0.033s (30.000 fps)
            Interval: Discrete 0.042s (24.000 fps)
            Interval: Discrete 0.050s (20.000 fps)
```

**Comment by Gatsby23** [comment]
> The Insta360 Air supports the following formats:
> 
> ```
>     [0]: 'MJPG' (Motion-JPEG, compressed)
>         Size: Discrete 1472x736
>             Interval: Discrete 0.033s (30.000 fps)
>         Size: Discrete 2176x1088
>             Interval: Discrete 0.033s (30.000 fps)
>         Size: Discrete 3008x1504
>             Interval: Discrete 0.033s (30.000 fps)
>     [1]: 'H264' (H.264, compressed)
>         Size: Discrete 1920x960
>             Interval: Discrete 0.033s (30.000 fps)
>             Interval: Discrete 0.042s (24.000 fps)
>             Interval: Discrete 0.050s (20.000 fps)
>         Size: Discrete 2560x1280
>             Interval: Discrete 0.033s (30.000 fps)
>             Interval: Discrete 0.042s (24.000 fps)
>             Interval: Discrete 0.050s (20.000 fps)
> ```

Thank you for your quick replay. Can it link to the PC through ROS directly ? I wanna to use it with an extra IMU, could you please give me some help? Thank you very much!

**Comment by Martin-Oehler** [comment]
Yes. Please see my previous reply in this thread.

**Comment by Gatsby23** [comment]
> Yes. Please see my previous reply in this thread.

Thank you very much! I get your point! That's really meanful to me. Thank you for your help at all!

**Comment by Gatsby23** [comment]
> The Insta360 Air supports the following formats:
> 
> ```
>     [0]: 'MJPG' (Motion-JPEG, compressed)
>         Size: Discrete 1472x736
>             Interval: Discrete 0.033s (30.000 fps)
>         Size: Discrete 2176x1088
>             Interval: Discrete 0.033s (30.000 fps)
>         Size: Discrete 3008x1504
>             Interval: Discrete 0.033s (30.000 fps)
>     [1]: 'H264' (H.264, compressed)
>         Size: Discrete 1920x960
>             Interval: Discrete 0.033s (30.000 fps)
>             Interval: Discrete 0.042s (24.000 fps)
>             Interval: Discrete 0.050s (20.000 fps)
>         Size: Discrete 2560x1280
>             Interval: Discrete 0.033s (30.000 fps)
>             Interval: Discrete 0.042s (24.000 fps)
>             Interval: Discrete 0.050s (20.000 fps)
> ```

Sorry, there are only last question bothering me a lot. Does the image from the insta360 air have latency of played as same as a normal pinhole camera?

**Comment by Martin-Oehler** [comment]
As the Insta360 Air communicates via USB as an UVC device as other cameras as well, there is no difference in latency.

**Comment by johnzjq** [comment]
> Yes. Please see my previous reply in this thread.

Hi Martin, is it possible to capture the IMU measurement together with the images?

**Comment by Martin-Oehler** [comment]
As far as I am aware, the Insta360 Air does not have an integrated IMU.

**Comment by johnzjq** [comment]
> As far as I am aware, the Insta360 Air does not have an integrated IMU.

I see. But have you tried other products from Insta360? The insta360 one series do have IMU installed.

**Comment by StefanFabian** [comment]
No, unfortunately, we do not have experience with newer Insta360 products.
If you investigate their application in robotics, we would be very interested in your findings.

**Comment by johnzjq** [comment]
> No, unfortunately, we do not have experience with newer Insta360 products. If you investigate their application in robotics, we would be very interested in your findings.

Thanks! Stefan! I will try to contact insta360 and see if it is possible to have a trial on their product!

**Comment by avicomac** [comment]
> > No, unfortunately, we do not have experience with newer Insta360 products. If you investigate their application in robotics, we would be very interested in your findings.
> 
> Thanks! Stefan! I will try to contact insta360 and see if it is possible to have a trial on their product!

Hi,May I ask if you have made any progress on this issue? I am currently considering using the built-in IMU of Insta360 x2 to implement vio. Can you give me some suggestions?
